### PR TITLE
logging - expose undefined field, multiline parameters

### PR DESCRIPTION
**Pull request by mburke5678**
https://github.com/openshift/openshift-ansible/pull/11991

### CODE REVIEW

**Comment by mburke5678** [comment]
@richm PTAL. 
Might be my ignorance but the defaults listed here: https://github.com/openshift/openshift-ansible/pull/11991/files#diff-3030bf2d7977c367aa6020568e1d9dc5 don't always match what you have in the README.

**Comment by richm** [comment]
> @richm PTAL.
> Might be my ignorance but the defaults listed here: https://github.com/openshift/openshift-ansible/pull/11991/files#diff-3030bf2d7977c367aa6020568e1d9dc5 don't always match what you have in the README.

Right - they are intentionally *not* the default values - it is an example of how to customize those values.  The way to have logging deployed with the default values is to *not* set those parameters in your ansible playbook, and just let fluentd use the defaults baked into the image.

**Comment by mburke5678** [comment]
@anpingli PTAL

**Review by anpingli — COMMENTED** [review]
LGTM

File: install_config/aggregate_logging.adoc [hunk 1]
```diff
@@ -363,6 +363,59 @@ variable allows the EFK to watch the specified audit log file or the
...
+|`openshift_logging_fluentd_undefined_max_num_fields`
+|Specify a limit to the number of undefined fields when using 
+`openshift_logging_fluentd_merge_json_log`. Logs can contain hundreds of undefined fields, 
+which causes problems with Elasticsearch.  If there are more than the specified 
+number of fields, the fields will be converted into a JSON hash string and 
+stored in the `openshift_logging_fluentd_undefined_name` field.  
+The default value is `-1` which means an unlimited number of fields.
+
+|`openshift_logging_fluentd_use_multiline_json`
+|Set to true to force Fluentd to reconstruct any split log lines into a single line.
```
Comment (adellape) [inline install_config/aggregate_logging.adoc]: Mark up `true`.

Also the period at the end of this line is in the middle of a sentence when rendered.

File: install_config/aggregate_logging.adoc [hunk 2]
```diff
@@ -363,6 +363,59 @@ variable allows the EFK to watch the specified audit log file or the
...
+
+|`openshift_logging_fluentd_use_multiline_json`
+|Set to true to force Fluentd to reconstruct any split log lines into a single line. 
+when using `openshift_logging_fluentd_merge_json_log`. With the `json-file` 
+driver, Docker splits log lines at a size of 16k bytes. The default is `false`.
+
+|`openshift_logging_fluentd_use_multiline_journal`
+|Set to `true` to force Fluentd to reconstruct the split lines into a single line
+when using `openshift_logging_fluentd_merge_json_log`. With the `json-file` driver, 
+Docker splits log lines at a size of 16k bytes. The default is `false`.
```
Comment (adellape) [inline install_config/aggregate_logging.adoc]: This entry for `openshift_logging_fluentd_use_multiline_journal` looks the same as the entry for `openshift_logging_fluentd_use_multiline_json`?

**Review by adellape — APPROVED** [review]
Looks good, a few nits.

File: install_config/aggregate_logging.adoc (hunk 2 above)
Comment (richm) [reply install_config/aggregate_logging.adoc]: Good catch - it should say `With the `journald` driver,...`

**Pull request merged**